src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/ConnectionPool/HttpConnectionPool.Http3.cs

```diff
@@ -87,19 +87,18 @@ internal sealed partial class HttpConnectionPool
                         ThrowGetVersionException(request, 3, reasonException);
                     }
 
-                    long queueStartingTimestamp = HttpTelemetry.Log.IsEnabled() || Settings._metrics!.RequestsQueueDuration.Enabled ? Stopwatch.GetTimestamp() : 0;
-                    Activity? waitForConnectionActivity = ConnectionSetupDistributedTracing.StartWaitForConnectionActivity(authority);
+                    WaitForHttp3ConnectionActivity waitForConnectionActivity = new WaitForHttp3ConnectionActivity(Settings._metrics!, authority);
 
                     if (!TryGetPooledHttp3Connection(request, out Http3Connection? connection, out http3ConnectionWaiter))
                     {
+                        waitForConnectionActivity.Start();
```

Comment:
Could we use `WaitForConnectionWithTelemetryAsync`? That's what I meant by making it analogous to H2.

Reply:
`WaitForConnectionWithTelemetryAsync` only covers the connection establishment. We could refactor the H3 code so it that it also includes the opening of the first stream, but it still wouldn't help us since follow up requests on the same connection still may end up waiting in `OpenOutboundStreamAsync`, which is a case that doesn't fit the `WaitForConnectionWithTelemetryAsync` model at all.

Long story short, I don't see how would this be possible, and I don't think it's worth to chase this goal.